### PR TITLE
Enable and fix await-only-futures lint

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

**Comment by kevmoo** [comment]
The SDK constraint change will be subsumed by https://github.com/dart-lang/test/pull/638

File: pubspec.yaml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 name: test
-version: 0.12.22
+version: 0.12.23-dev
 author: Dart Team <misc@dartlang.org>
 description: A library for writing dart unit tests.
 homepage: https://github.com/dart-lang/test
 environment:
-  sdk: '>=1.14.0 <2.0.0'
+  sdk: '>=1.22.1 <2.0.0'
```
Comment (grouma) [inline pubspec.yaml]: Should we just wait for https://github.com/dart-lang/test/pull/638 to go through? It'll be less tests to run too :D

Comment (kevmoo) [inline pubspec.yaml]: Sure – just hurry up! 😁

Comment (grouma) [inline pubspec.yaml]: Waiting for @nex3 to provide the LGTM.

**Comment by kevmoo** [comment]
> I'm wondering if we should add a lint for unawaited futures as well.

@grouma that lint exists, but there are a LOT of false positives in pkg/test

You can exclude these one-by-one with line-level comments, thought.

Worth looking into

**Comment by kevmoo** [comment]
@nex3 want me should wait for Travis, or we feel okay about this?

**Comment by nex3** [comment]
I think we're fine to just land it, since Travis passed before the merge.

**Pull request merged**